client/state/inline-help/test/selectors.js

```diff
@@ -135,3 +138,93 @@ describe( '#getInlineHelpSearchResultsForQuery()', () => {
 		expect( getInlineHelpSearchResultsForQuery( state, 'foo' ) ).to.deep.equal( [] );
 	} );
 } );
+
+describe( 'Admin section search and filters', () => {
+	describe( 'getAdminSectionsResults()', () => {
+		test( 'should return `[]` when there is no search term', () => {
+			const result = getAdminHelpResults( {} );
+			expect( result ).to.deep.equal( [] );
+		} );
+
+		test( 'should return admin help results for `domain` term', () => {
+			const results = getAdminHelpResults( {}, 'Add a new domain' );
+			expect( results ).to.be.a( 'array' );
+			expect( results ).to.not.deep.equal( [] );
+			expect( results ).to.not.have.length( 0 );
+		} );
+	} );
+
+	describe( 'filterListBySearchTerm()', () => {
```

Comment:
This test for `filterListBySearchTerm ` is more of a unit test whereas you other tests for `getAdminSectionsResults` more of an integration test (as `getAdminSectionsResults` depends on `filterListBySearchTerm`). 

Ultimately the feature we want to test is whether `getAdminSectionsResults` returns the correct results. 
I see `filterListBySearchTerm` as an implementation detail which we should be able to swap out without breaking  `getAdminSectionsResults`. 

For this reason, it feels like our tests would be more **resilient to change** if they simply asserted on the `getAdminSectionsResults` method rather than on the implementation details. This why if the implementation changes, the tests don't have to change as well. That lowers maintenance and improves confidence.

Personally I don't think `filterListBySearchTerm` should be exported by the module as part of the API. We may wish to swap it out in future and making it public at this point invites it's use as a public method.

Reply:
I would say that I think there's value in testing `filterListBySearchTerm` in isolation as a utility or helper function.

However, I think we're using the results of these tests to also assert that `getAdminSectionsResults` is working correctly. I think we're leaving `getAdminSectionsResults` undertested and it's provided a false confidence.

For example, I would easily switch out the implementation of `getAdminSectionsResults` for simple logic which passes the tests for `getAdminSectionsResults` but would not cover all our use cases.